go/chat/boxer_test.go

```diff
@@ -1488,3 +1499,71 @@ func TestRemarshalBoxed(t *testing.T) {
 
 	require.Equal(t, boxed1.ClientHeader.OutboxID, boxed2.ClientHeader.OutboxID, "obids should have same value")
 }
+
+func randomTeamEK() keybase1.TeamEk {
+	randBytes, err := libkb.RandBytes(32)
+	if err != nil {
+		panic(err)
+	}
+	seed := libkb.MakeByte32(randBytes)
+	dhKey := (*ephemeral.TeamEKSeed)(&seed).DeriveDHKey()
+	teamEK := keybase1.TeamEk{
+		Seed: seed,
+		Metadata: keybase1.TeamEkMetadata{
+			Kid: dhKey.GetKID(),
+		},
+	}
+	return teamEK
+}
+
+func TestExplodingMessageUnbox(t *testing.T) {
+	key := cryptKey(t)
+	ephemeralKey := randomTeamEK()
+	text := "hello exploding"
+	tc, boxer := setupChatTest(t, "exploding")
+	defer tc.Cleanup()
+	// We need a user for unboxing to work.
+	u, err := kbtest.CreateAndSignupFakeUser("unbox", tc.G)
+	if err != nil {
+		t.Fatal(err)
+	}
+	msg := textMsgWithSender(t, text, gregor1.UID(u.User.GetUID().ToBytes()), chat1.MessageBoxedVersion_V3)
+
+	// Set the ephemeral metadata, to indicate that the messages is exploding.
+	msg.ClientHeader.EphemeralMetadata = &chat1.MsgEphemeralMetadata{
+		Lifetime: 99999,
```

Comment:
i just landed https://github.com/keybase/client/pull/11439 so you should rebase to get the changes